### PR TITLE
Use newManager instead of withManager

### PR DESCRIPTION
**Pull request by glasserc**
withManager is deprecated in current http-client.

### CODE REVIEW

File: src/Network/Eureka/Connection.hs [hunk 1]
```diff
@@ -30,8 +30,8 @@ import           Network.BSD               (getHostName)
 import           Network.HTTP.Client       (HttpException,
                                             Manager, Request (checkStatus, method, requestBody, requestHeaders),
                                             RequestBody (RequestBodyLBS),
-                                            defaultManagerSettings, queryString,
-                                            responseStatus, withManager,
+                                            defaultManagerSettings,  newManager,
```
Comment (potomak) [inline src/Network/Eureka/Connection.hs]: OCD: there are two spaces instead of one.

Comment (glasserc) [inline src/Network/Eureka/Connection.hs]: Wow, you have an eagle eye! Fixed in 91f1426c39dd9e44dce94c3288dfe59ac0e6c7fa.

**Comment by potomak** [comment]
I have a question: has always been like this (that you can safely avoid to close an HTTP connection manager) or there's a minimum version we should enforce for http-client?

**Comment by andrewchizhik** [comment]
Well, closeManager was deprecated in [http-client version 0.4.18](https://github.com/snoyberg/http-client/commit/32b02a78d6cacc2884bc52a8787c2460b13e06f9); withManager was deprecated before that. Seems like 0.4.18 would be a safe minimum version.

**Comment by glasserc** [comment]
Added a minimum version in 9b560b151b2bca9fbab5f112ec4d3538de30c574.

**Pull request merged**